### ISSUE TITLE
Auto-resizing windows are fully stretched when first apparing in a viewport with negative coordinates

### ISSUE THREAD

**Issue by DimaKoltun** [ocornut/imgui#5215]
**Version/Branch of Dear ImGui:**

Version: 1.88 (WIP)
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_sdl.cpp + imgui_impl_opengl3.cpp
Operating System: Windows 10

**My Issue/Question:**

I have 2 monitors with next setup:
![image](https://user-images.githubusercontent.com/83829508/163983505-cd8b7fc0-3b9f-4dc9-b5c4-43e30ef79494.png)
Reproduce case:
1) Remove imgui.ini
2) Move app to second window
3) Open any modal window, for example, from examples: "Stacked modals.."
Result:
Modal window is fully streched.
It seems, that this bug happens, because window->DC.IdealMaxPos isn't initialized on CreateWindow(), that used in CalcWindowContentSizes()

**Screenshots/Video**

https://user-images.githubusercontent.com/83829508/163986238-7c2cf7a9-02a1-442c-b9d9-3048d38d9119.mp4

**Comment by chreden** [comment]
A user reported this to me coincidentally today as well. To reproduce this the monitor on the right in the layout has to have "Make this my main display" checked.

**Comment by ocornut** [comment]
Thank you for reporting this. It indeed looks like a bug, will investigate soon.

**Comment by ocornut** [comment]
I don't have a repro right now but my intuition is that it a bug that triggers on negative absolute coordinates.
Out of curiosity can you try to move Monitor 2 to the _right side_ of Monitor 1 and see if it makes a difference ? 
That would confirm or not my theory.

**Comment by ocornut** [comment]
> To reproduce this the monitor on the right in the layout has to have "Make this my main display" checked.

Ah! amusingly I had skipped over this message.. That confirms it. Sorry for missing this :)

**Comment by ocornut** [comment]
In `CreateNewWindow()` can you indeed modify the line that says:
```
window->DC.CursorStartPos = window->DC.CursorMaxPos = window->Pos;
```
turning it into:
```
window->DC.CursorStartPos = window->DC.CursorMaxPos = window->IdealMaxPos = window->Pos;
```
And confirm that it would fix the bug?

**Comment by ocornut** [comment]
Managed to craft a repro and confirmed the fix. 
Pushed fix b5b704d. @DimaKoltun I took the liberty to give you attribution for the commit since your description nailed the cause of the bug.

Bug got added in 6af92b05

Sorry for the multiple messages :)
